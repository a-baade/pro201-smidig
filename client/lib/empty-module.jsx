/* For tests. DonÂ´t delete it */