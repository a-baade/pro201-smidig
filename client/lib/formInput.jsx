import React from "react";

export function FormInput({ value, label, onChangeValue }) {
  return (
    <div className={"form-input"}>
      <label>
        <strong>{label}</strong>{" "}
        <input value={value} onChange={(e) => onChangeValue(e.target.value)} />
      </label>
    </div>
  );
}

export function FormInputPassword({ value, label, onChangeValue }) {
  return (
    <div className={"form-input"}>
      <label>
        <strong>{label}</strong>{" "}
        <input type={"password"} value={value} onChange={(e) => onChangeValue(e.target.value)} />
      </label>
    </div>
  );
}
