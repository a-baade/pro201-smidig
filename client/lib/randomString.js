export function randomString(length) {
  const possible =
    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
  let result = "";
  for (let i = 0; i < length; i++) {
    result += possible.charAt(Math.floor(Math.random() * possible.length));
  }
  return result;
}
