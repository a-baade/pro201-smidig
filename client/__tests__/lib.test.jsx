import React from "react";
import ReactDOM from "react-dom";
import { Link, MemoryRouter } from "react-router-dom";
import { FormInput, FormInputPassword } from "../lib/formInput";

describe("FormInput and FormPassword components", () => {
  it("should test form input", function () {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <FormInput />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchInlineSnapshot(
      `"<div class=\\"form-input\\"><label><strong></strong> <input value=\\"\\"></label></div>"`
    );
    expect(element.innerHTML).toMatchSnapshot();
  });

  it("should test form password", function () {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <FormInputPassword />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchInlineSnapshot(
      `"<div class=\\"form-input\\"><label><strong></strong> <input type=\\"password\\" value=\\"\\"></label></div>"`
    );
    expect(element.innerHTML).toMatchSnapshot();
  });
});
