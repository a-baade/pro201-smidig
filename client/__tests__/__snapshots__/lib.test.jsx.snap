// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput and FormPassword components should test form input 2`] = `"<div class=\\"form-input\\"><label><strong></strong> <input value=\\"\\"></label></div>"`;

exports[`FormInput and FormPassword components should test form password 2`] = `"<div class=\\"form-input\\"><label><strong></strong> <input type=\\"password\\" value=\\"\\"></label></div>"`;

exports[`FormInput and FormPassword components should test loading spinner 2`] = `"<div class=\\"loading-wrapper\\"><div class=\\"loading\\"><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div></div>"`;
